src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -703,105 +708,110 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
                     break;
                 }
+
                 case 3:
+                case 4:
```

Comment:
hhm, any reason why we are combining 3 and 4? This can lead to accidently not handling things correctly in future.

Reply:
If the intent is to share the following code, then we should extract it out I think in a common method and call it for case 3 and 4.

```c#
if (intrin.op2->IsVectorZero())
{
  ...
}
else
{
  ...
}
```

Reply:
That was my intent; I'll do that